generation/core/generator.js

```diff
@@ -219,15 +220,21 @@ module.exports = function getGenerator({
       const json = isObjectiveC ? objectiveCParser(input) : javaMethodParser(input);
 
       // set default name
+      const pathFragments = outputFile.split('/');
       if (!json.name) {
-        const pathFragments = outputFile.split('/');
         json.name = pathFragments[pathFragments.length - 1].replace('.js', '');
       }
       const ast = t.program([createClass(json), createExport(json)]);
       const output = generate(ast);
 
       const commentBefore = '/**\n\n\tThis code is generated.\n\tFor more information see generation/README.md.\n*/\n\n';
 
+      pathFragments.pop(); // remove filename
+      const outputPath = pathFragments.join("/")
+      const absoluteUtilsPath = path.resolve('../detox/src/utils');
+      const relativeUtilsPath = path.relative(outputPath, absoluteUtilsPath);
+      const logImport = `const log = require('${relativeUtilsPath}/logger');`;
```

Comment:
Change to `const log = require('${relativeUtilsPath}/logger').child({ __filename });`, so we can track from which files the log messages come.

Reply:
Also, could you extract this logic to a separate function, similar to a function `createTypeCheck(...)` above? I see that `generator` function looks already loong enough, why make it even bigger? :-)

Reply:
Sure, no problem 👍 I think there are some parts I would like to slim down a bit and structure better, will go on with this soonish. My first goal was to get it working,so that I can see every edge case I ran into and make a better layer of abstraction later on ;)